eth/core/state/interfaces.go

```diff
@@ -101,6 +101,8 @@ type (
 		Logs() []*coretypes.Log
 		// GetLogs returns the logs of the tx with the given metadata.
 		GetLogs(hash common.Hash, blockNumber uint64, blockHash common.Hash) []*coretypes.Log
+		// GetBlockLogsAndClear returns the logs of the block and clears the journal.
+		GetBlockLogsAndClear(blockHash common.Hash) []*coretypes.Log
```

Comment:
we need to stop adding shit not in geth keke

Reply:
better separation of concerns to have the block logs list in the logs journal imo, especially based on our abstractions. geth stores logs per block as well

Reply:
why do we need an Clear then?

just clear on StateDB Reset

Reply:
That's how Geth does it tho lmeow

Reply:
lmao wow geth never clears the logs map

Reply:
we can also not clear but I think its a good optimization